Board.java

```diff
@@ -13,15 +15,27 @@ public String getState() {
         return this.state;
     }
 
+     public Boolean availablePosition(int pos) {
+        if (state.charAt(pos) == '-' && pos >= 0 && pos <= dimension) {
```

Comment:
don't use abbreviations like '_pos_', always use descriptive words like '_position_'.  
We do this because it might not only be the original author of the code who needs to maintain it.  The next developer who came along might not know what you were thinking with '_pos_' or '_ply_': could you have meant 'positive'?  It's fairly obvious in this case, but with _ply_ it took me longer than a millisecond to understand.

Then, I would change the order of your conditional as it seems more logical to check if the _position_ is valid before checking the char at that _position_. So I would make it:
if (position >= 0 && position < dimension && state.charAt(position) == '-')

Having said that, what happens if _position_ is -1 or greater than '_dimension_' and you try to do _state.charAt(-1)_ ?   **Have you got a test for that?**
If that throws an exception then there is a bigger problem, but if it evaluates _state.charAt(-1) == '-'_ to be false then the other two conditions aren't necessary.